doc/design/model_zoo.md

```diff
@@ -0,0 +1,174 @@
+
+It is necessary to record the content or the MD5 hash of the training submitter program in the model zoo table for experiment management. Please be aware that the training submitter encodes all three categories of hyperparameters, as listed in the above sections.  Suppose that the analyst re-trains the model with different hyperparameter settings, the training submitter changes accordingly, and SQLFlow should be able to remind the analyst to either uses a new model ID or overwrites the existing row in the model zoo table.
+
+### Model Sharing
+
+After all, what is a model zoo? A model zoo refers to all the model definitions and trained models accessible by a deployment of SQLFlow.  It contains one or more model definition Docker images and the source code repositories that build the images.  It also includes the model zoo table configured to work with all submitter programs generated by the deployment.
+
+Within a deployment, it is straightforward to share a trained model.  If the analyst, `an_analyst`, in the above example wants to use her own trained model `my_first_model` for prediction, she could use the short name `my_first_model`.
+
+```sql
+SELECT ... TO PREDICT ... USING my_first_model
+```
+
+If another analyst wants to use the trained model, he would need to use the full name as `an_analyst/my_first_model`.
+
+```sql
+SELECT ... TO PREDICT ... USING an_analyst/my_first_model
+```
+
+There could be more than one model definitions in each model's Docker image. We need to be able to find out which model definition is used to train current saved model. Also if we only want to reuse the model definition to train a new model we need to know the model definition class name, so that we can pass it to the `TRAIN` clause.
+
+To list all trained models of one user, you can do:
+
+```sql
+SQLFLOW LIST an_analyst
+```
+
+This should output a table showing the saved models and which model definition it was using.
+
+```
+|  saved model    |   model def    |
+| my_first_model  | MyDNNRegressor |
+| my_second_model | MyDNNRegressor |
+```
+
+Within **any** deployment that have internet access, to list published models:
+
+```sql
+SQLFLOW LIST [models.sqlflow.com/an_analyst]
+```
+
+Display model definitions and documentation of the published model:
+
+```
+SQLFLOW DESCRIBE model.sqlflow.com/an_analyst/my_first_model;
+| available model defs |
+| MyDNNRegressor       |
+| MyDNNClassifier      |
+
+SQLFLOW DESCRIBE model.sqlflow.com/an_analyst/my_first_model.MyDNNRegressor;
+
+Documatation for my_first_model.MyDNNRegressor
+...
+...
+```
+
+Use a published model to predict some data:
+
+```sql
+SELECT ... TO PREDICT employee.predicted_salary USING model.sqlflow.com/an_analyst/my_first_model
```

Comment:
Do we need a `docker pull` or `git pull` like command here, to pull the model in the model zoo to local?

Reply:
The model zoo is a collection of Docker images. When the SQLFlow server runs `docker run`, the Docker client would pull if needed.